src/Core/Module/Parser/ModuleParser.php

```diff
@@ -0,0 +1,415 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShop\PrestaShop\Core\Module\Parser;
+
+use PhpParser\Error;
+use PhpParser\Node;
+use PhpParser\Node\Expr;
+use PhpParser\NodeDumper;
+use PhpParser\NodeFinder;
+use PhpParser\Parser;
+use PhpParser\ParserFactory;
+
+/**
+ * This parser scan the PHP code of a module main class and extracts information from
+ * static PHP analysis, so it is always up-to-date with the raw PHP content.
+ *
+ * It was done previously by using include($moduleFile) and accessing data via $module->version,
+ * the problem is that ones the class has been included it remains in the same state during the
+ * whole process, providing it from being updated even if the file has been changed, which can lead
+ * to unexpected values during an upgrade process. This method ensure we always get the most
+ * up-to-date value from the PHP files.
+ */
+class ModuleParser
+{
+    private Parser $phpParser;
+
+    private NodeFinder $nodeFinder;
+
+    private const DEFAULT_EXTRACTED_PROPERTIES = [
+        'name',
+        'tab',
+        'version',
+        'ps_versions_compliancy',
+        'author',
+        'displayName',
+        'description',
+        'author_uri',
+        'dependencies',
+        'description_full',
+        'is_configurable',
+        'need_instance',
+        'limited_countries',
+        'hooks',
```

Comment:
@jolelievre what's the idea behind adding this property? it's not a default one

Reply:
@jolelievre Remember that many modules (including my "put content anywhere") use magic methods, so don't trust this hook list. :-)